app/allocations/middleware/set-results.allocations.test.js

```diff
@@ -34,40 +38,61 @@ describe('Allocations middleware', function() {
 
     context('when service resolves', function() {
       beforeEach(async function() {
-        allocationService.getByDateAndLocation.resolves(mockActiveMoves)
+        allocationService.getByDateAndLocation
+          .onCall(0)
+          .resolves(mockActiveMoves)
+        allocationService.getByDateAndLocation
+          .onCall(1)
+          .resolves(mockCancelledMoves)
         await middleware(req, res, next)
       })
 
       it('should call the data service with request body', function() {
+        expect(allocationService.getByDateAndLocation).to.have.been.calledTwice
         expect(
-          allocationService.getByDateAndLocation
-        ).to.have.been.calledOnceWithExactly({
-          status: 'proposed',
-          moveDate: ['2019-01-01', '2019-01-07'],
-          fromLocationId: '123',
-        })
+          allocationService.getByDateAndLocation.firstCall.args
+        ).to.deep.equal([
+          {
+            status: 'proposed',
+            moveDate: ['2019-01-01', '2019-01-07'],
+            fromLocationId: '123',
+          },
+        ])
+        expect(
+          allocationService.getByDateAndLocation.secondCall.args
+        ).to.deep.equal([
+          {
+            status: 'cancelled',
+            moveDate: ['2019-01-01', '2019-01-07'],
+            fromLocationId: '123',
+          },
+        ])
       })
 
       it('should set results on req', function() {
         expect(req).to.have.property('results')
         expect(req.results).to.deep.equal({
           active: mockActiveMoves,
-          cancelled: [],
+          cancelled: mockCancelledMoves,
         })
       })
 
       it('should set resultsAsTable on req', function() {
         expect(req).to.have.property('resultsAsTable')
         expect(req.resultsAsTable).to.deep.equal({
           active: mockActiveMoves,
-          cancelled: [],
+          cancelled: mockCancelledMoves,
         })
       })
 
       it('should call allocationsToTable presenter', function() {
-        expect(presenters.allocationsToTable).to.be.calledOnceWithExactly(
+        expect(presenters.allocationsToTable).to.be.calledTwice
+        expect(presenters.allocationsToTable.firstCall.firstArg).to.deep.equal(
```

Comment:
It might actually be nice to split these tests out. It's a bit easier to debug with trying to keep an approach of one assertion to an `it` block. So you could have:

```javascript
it('should call allocationsToTable presenter correct number of times', function() {
  ...
})

it('should call allocationsToTable presenter for active moves', function() {
  ...
})

it('should call allocationsToTable presenter for cancelled moves', function() {
  ...
})
```